manifest.go

```diff
@@ -156,6 +158,50 @@ func validateManifest(s string) ([]error, error) {
 	return warns, nil
 }
 
+// validateProjectRoot validates ProjectRoot. It it used as a goroutine for
+// concurrent validation of ProjectRoot(s).
+func validateProjectRoot(pr gps.ProjectRoot, sm gps.SourceManager, wg *sync.WaitGroup, errorCh chan<- error) {
+	defer wg.Done()
+	origPR, err := sm.DeduceProjectRoot(string(pr))
+	if err != nil {
+		errorCh <- err
+	} else if origPR != pr {
+		errorCh <- fmt.Errorf("the name for %q should be changed to %q", pr, origPR)
+	}
+}
```

Comment:
Ah, this works, but I actually meant you could define a variable to hold the closure which you were already using, rather than invoking it in place. Then the function doesn't have to expand to include those 3 extra arguments.

Reply:
ah! first class functions 😊 Haven't used them in that way much. Anyway, now I know how to use `waitGroup` with an external function 😀

Changing to hold the function in a variable.

Thanks again!